src/graph.h

```diff
@@ -343,4 +348,32 @@ struct DependencyScan {
   DyndepLoader dyndep_loader_;
 };
 
+// Prioritize edges by largest critical time first
+struct EdgePriorityCompare {
+  bool operator()(const Edge* e1, const Edge* e2) const {
+    const int64_t ct1 = e1->critical_time();
+    const int64_t ct2 = e2->critical_time();
+    if (ct1 != ct2) {
+      return ct1 < ct2;
+    }
+    return e1->id_ < e2->id_;
+  }
+};
+
+// Set of ready edges, sorted by priority
```

Comment:
nit: a priority queue is not a set (i.e. it can contain duplicates, which can have surprising consequences). Also please clarify "sorted by priority" because this is very subtle. For example, what about:

```
A priority queue holding non-owning Edge pointers, its pop() method returns the edge with the largest ciritical_time() in the queue, and largest id() otherwise.
```

Also please call this EdgePriorityQueue or EdgeCriticalTimePriorityQueue for clarity.